$MirOS: ports/emulators/qemu/snapshot/patches/patch-Makefile_target,v 1.1 2008/05/07 15:53:08 tg Exp $
$OpenBSD: patch-Makefile_target,v 1.12 2008/04/28 22:52:38 todd Exp $

	• use scn.c due to systrace on /dev/null.gcda
	• use -D_NETBSD_SOURCE for some <math.h> funcs on MirBSD #10

--- Makefile.target.orig	Wed Sep 23 19:01:07 2009
+++ Makefile.target	Sat Oct 17 16:43:27 2009
@@ -24,7 +24,7 @@ PROGS=$(QEMU_PROG)
 # cc-option
 # Usage: CFLAGS+=$(call cc-option, $(CFLAGS), -falign-functions=0, -malign-functions=0)
 
-cc-option = $(shell if $(CC) $(1) $(2) -S -o /dev/null -xc /dev/null \
+cc-option = $(shell if $(CC) $(1) $(2) -S $(SRC_PATH)/scn.c \
               > /dev/null 2>&1; then echo "$(2)"; else echo "$(3)"; fi ;)
 
 HELPER_CFLAGS=
@@ -52,7 +52,7 @@ ifeq ($(ARCH),ia64)
 CFLAGS+=-mno-sdata
 endif
 
-CPPFLAGS+=-D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+CPPFLAGS+=-D_GNU_SOURCE -D_NETBSD_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 CPPFLAGS+=-U_FORTIFY_SOURCE
 LIBS+=-lm
 ifdef CONFIG_WIN32
